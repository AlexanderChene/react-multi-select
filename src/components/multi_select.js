import React, { PureComponent } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

import withMultiSelectState from "./multi_select_state";
import SourceList from "./source_list";
import DestinationList from "./destination_list";

import styles from "./multi_select.scss";
import Loader from "./loader/loader";

class MultiSelect extends PureComponent {
  static propTypes = {
    selectedItems: PropTypes.array,
    filteredItems: PropTypes.array,
    loading: PropTypes.bool,
    messages: PropTypes.object,
    onChange: PropTypes.func,
    showSearch: PropTypes.bool,
    showSelectAll: PropTypes.bool,
    searchIcon: PropTypes.string,
    deleteIcon: PropTypes.string,
    searchRenderer: PropTypes.func,
    selectedItemRenderer: PropTypes.any,
    height: PropTypes.number,
    itemHeight: PropTypes.number,
    loaderRenderer: PropTypes.any
  };

  static defaultProps = {
    selectedItems: [],
    filteredItems: [],
    messages: {},
    showSearch: true,
    showSelectAll: true,
    height: 400,
    itemHeight: 40,
    loaderRenderer: Loader
  };

  calculateHeight() {
    let { height, showSearch, showSelectAll, itemHeight } = this.props;
    height = showSearch ? height - 45 : height;
    height = showSelectAll ? height - itemHeight : height;
    return height;
  }

  render() {
    const {
      wrapperClassName,
      showSearch,
      height,
      itemHeight,
      showSelectAll,
      itemRenderer,
      selectedItemRenderer,
      filteredItems,
      selectedItems,
      getList,
      filterItems,
      selectAllItems,
      isAllSelected,
      clearAll,
      selectItem,
      unselectItem,
      searchIcon,
      searchRenderer,
      selectAllRenderer,
      selectionStatusRenderer,
      noItemsRenderer,
      loaderRenderer,
      messages,
      loading
    } = this.props;
    const calculatedHeight = this.calculateHeight();
    const selectedIds = selectedItems.map(item => item.id);
    const LoaderRenderer = loaderRenderer;
    return (
      <div
        className={classnames(styles.wrapper, wrapperClassName)}
        style={{ height }}
      >
        {loading && <LoaderRenderer />}
        {!loading && (
          <SourceList
            searchRenderer={searchRenderer}
            selectAllRenderer={selectAllRenderer}
            showSearch={showSearch}
            filterItems={filterItems}
            searchIcon={searchIcon}
            messages={messages}
            showSelectAll={showSelectAll}
            itemHeight={itemHeight}
            selectAllItems={selectAllItems}
            isAllSelected={isAllSelected}
            selectedIds={selectedIds}
            itemRenderer={itemRenderer}
            getList={getList}
            filteredItems={filteredItems}
            calculatedHeight={calculatedHeight}
            selectItem={selectItem}
            noItemsRenderer={noItemsRenderer}
          />
        )}
        {!loading && (
          <DestinationList
            selectionStatusRenderer={selectionStatusRenderer}
            selectedIds={selectedIds}
            clearAll={clearAll}
            messages={messages}
            selectedItems={selectedItems}
            itemHeight={itemHeight}
            height={height}
            unselectItem={unselectItem}
            selectedItemRenderer={selectedItemRenderer}
            noItemsRenderer={noItemsRenderer}
          />
        )}
      </div>
    );
  }
}

export default withMultiSelectState(MultiSelect);
