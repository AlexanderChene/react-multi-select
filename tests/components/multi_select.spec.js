import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { MultiSelect } from "../../src/components/multi_select";

const CustomComponent = jest
  .fn(() => <div>Custom Component</div>)
  .mockName("mockedComponent");

const custom_messages = {
  searchPlaceholder: "Find...",
  noItemsMessage: "No entries available...",
  noneSelectedMessage: "Nothing",
  selectedMessage: "Checked",
  selectAllMessage: "Check all",
  clearAllMessage: "Uncheck all"
};

const selectAllItems = jest.fn().mockName("selectAllItems");
const filterItems = jest.fn().mockName("filterItems");
const selectItem = jest.fn().mockName("selectItem");
const unselectItems = jest.fn().mockName("unselectItems");
const clearAll = jest.fn().mockName("clearAll");

describe("MultiSelect", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect />);
    expect(tree).toMatchSnapshot();
  });

  test("custom selectAllRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectAllRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom searchRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect searchRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom noItemsRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect noItemsRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom itemRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect itemRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom messages", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect messages={custom_messages} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedIds", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectedIds={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectedItems={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passes disabled if maxSelectedItem has passed", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectedItems={[1, 2]} maxSelectedItems={2} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("does not pass disabled if maxSelectedItem has passed", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectedItems={[1, 2]} maxSelectedItems={4} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed selectAllItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectAllItems={selectAllItems} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed filterItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectAllItems={filterItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectItem", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectAllItems={selectItem} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed unselectItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect unselectItems={unselectItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed clearAll", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect clearAll={clearAll} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove select all", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSelectAll={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove search", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSearch={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("displays Loader when loading", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect loading={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("displays custom Loader when loading", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect loading={true} loaderRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("can remove selected items", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSelectedItems={false} />);
    expect(tree).toMatchSnapshot();
  });
});
