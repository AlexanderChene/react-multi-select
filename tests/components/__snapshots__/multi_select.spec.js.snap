// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect can remove search 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={false}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect can remove select all 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={355}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={false}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect can remove selected items 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect custom itemRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[MockFunction mockedComponent]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect custom messages 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={
      Object {
        "clearAllMessage": "Uncheck all",
        "noItemsMessage": "No entries available...",
        "noneSelectedMessage": "Nothing",
        "searchPlaceholder": "Find...",
        "selectAllMessage": "Check all",
        "selectedMessage": "Checked",
      }
    }
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={
      Object {
        "clearAllMessage": "Uncheck all",
        "noItemsMessage": "No entries available...",
        "noneSelectedMessage": "Nothing",
        "searchPlaceholder": "Find...",
        "selectAllMessage": "Check all",
        "selectedMessage": "Checked",
      }
    }
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect custom noItemsRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[MockFunction mockedComponent]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[MockFunction mockedComponent]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect custom searchRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[MockFunction mockedComponent]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect custom selectAllRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[MockFunction mockedComponent]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect default snapshot 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect displays Loader when loading 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <Loader />
</div>
`;

exports[`MultiSelect displays custom Loader when loading 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <mockConstructor />
</div>
`;

exports[`MultiSelect does not pass disabled if maxSelectedItem has passed 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={false}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed clearAll 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={[MockFunction clearAll]}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed filterItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={[MockFunction filterItems]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed selectAllItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={[MockFunction selectAllItems]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed selectItem 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={[MockFunction selectItem]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed selectedIds 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed selectedItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passed unselectItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={[MockFunction unselectItems]}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect passes disabled if maxSelectedItem has passed 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={true}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={false}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect will pass itemHeight 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={295}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={60}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={undefined}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={60}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect will pass selectAllHeight 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={60}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={40}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={60}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect will pass selectAllHeight without itemHeight 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={335}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    searchIcon={undefined}
    searchRenderer={[Function]}
    searchValue={undefined}
    selectAllHeight={20}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
    withGrouping={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
    withGrouping={false}
  />
</div>
`;

exports[`MultiSelect with responsiveHeight 1`] = `
<ReactHeight
  dirty={true}
  getElementHeight={[Function]}
  hidden={false}
  onHeightReady={[Function]}
  style={
    Object {
      "height": "70%",
    }
  }
>
  <_class
    filterFunction={[Function]}
    height={400}
    items={Array []}
    selectedItems={Array []}
  />
</ReactHeight>
`;

exports[`MultiSelect without responsiveHeight 1`] = `
<ReactHeight
  dirty={true}
  getElementHeight={[Function]}
  hidden={false}
  onHeightReady={[Function]}
  style={
    Object {
      "height": "400px",
    }
  }
>
  <_class
    filterFunction={[Function]}
    height={400}
    items={Array []}
    selectedItems={Array []}
  />
</ReactHeight>
`;
